Kitodo/src/main/java/org/kitodo/services/data/TaskService.java

```diff
@@ -117,7 +120,18 @@ private void manageProcessingUserDependenciesForIndex(Task task) throws CustomRe
             }
         } else {
             User user = task.getProcessingUser();
-            serviceManager.getUserService().saveToIndex(user);
+            if (user != null) {
+                serviceManager.getUserService().saveToIndex(user);
+            }
+            reSaveUserAfterRemoveFromProcessing(task);
+        }
+    }
+
+    private void reSaveUserAfterRemoveFromProcessing(Task task)
```

Comment:
It is not more a re-index after removing as your method does apply changes only to index?